reps/2023-04-28-remove-algorithms-from-rllib.md

```diff
@@ -0,0 +1,193 @@
+## Summary
+
+We'd like to remove aprox ~20 of the 30 algorithms from RLlib. We are considering doing this because 
+
+1. It would reduce the maintenance burden of RLlib.
+2. Many of these algorithms have successors that are more performant and/or easier to hyperparameter tune.
+3. The proposed algorithms have low ussage by the community according to our ray cluster telemetry readings.
+
+However, we recognize that the value that these algorithms have to the community, and so we are establishing the
+`rayproject/rllib-contrib` repo to house these algorithms. Each deprecated algorithm will have its own directory
+in this repo containing the following:
+
+1. The implementation of the algorithm.
+2. Any tests that are associated with ensuring the correctness of the algorithm.
+3. End to end example scripts that demonstrate how to run the algorithm
+4. The necessary requirements for running the algorithm
+5. boiler plate that allows the algorithm to be installed as a pip package.
+
+
+## Ussage
+
+Users will be able to replace the deprecated algorithms by doing the following:
+
+1. use git to clone `rayproject/rllib-contrib`
+2. navigate to the directory of the previously deprecated algorithm
+   for ex. `cd rllib-contrib/maml`
+3. run `pip install -e .` to install the algorithm as a pip package
+4. Import the algorithm, its policies and its config from that package instead of from `ray.rllib`
+
+In most usecases, the migration will look like replacing imports inside of an experiment script.
+
+For example:
+
+```python
+from gymnasium.wrappers import TimeLimit
+
+import ray
+from ray import air
+from ray import tune
+from ray.rllib.examples.env.cartpole_mass import CartPoleMassEnv
+from ray.tune.registry import register_env
+
+# from ray.rllib.algorithms.maml import MAML, MAMLConfig  # BEFORE
+from rllib_maml.maml import MAML, MAMLConfig  # AFTER
```

Comment:
can the pattern be something like `from rllib_contrib.maml import MAML`

Reply:
I want to know the estimate of work needed to allow this pattern. The answer could be no if the estimate of required work is too much.

Reply:
hmm I don't know how to make that happen without making all of the algorithms be under one python package.

Reply:
@avnishn @kouroshHakha What's the purpose that we need distribute the algorithms in different packages, instead of all in one package?

Reply:
If we distribute them into different packages, one potential risk is different packages will require different RLlib versions. For instance, rllib_a3c requries rllib<2.4, rllib_maml requries rllib>2.6. At least it's hard to maintain.

Reply:
hey @jovany-wang, that's one of the things that we actually want from this transition. i.e. for the algorithms to be pinned to a specific version of ray. Maintenance wise this will actually make our lives easier because we don't have to migrate all algorithms to the new APIs that we develop. The community can decide whether they want to upgrade them and contribute back.